### DISCUSSION TITLE
How to open a link as foreground new tab ?

### DISCUSSION THREAD

**Discussion by ghost** [gdh1995/vimium-c#724] [Q&A] [answered]
Shift+F opens links in new tab, but in background. I always had to navigate (shift+H) again & again.

How to open links in new tab & instantly bring up that tab ?

**Comment by gdh1995** [comment, marked as answer]
You may map a key to the command named `LinkHints.activateOpenInNewForegroundTab`

**Reply by ghost** [reply]
> You may map a key to the command named `LinkHints.activateOpenInNewForegroundTab`

Tq. `map <s-enter> LinkHints.activateOpenInNewForegroundTab` works for me.
But why isn't any key mapped to this bydefault ? It's supposed to be used by everyone frequently (more than the background opening)

**Comment by gdh1995** [comment]
Normally you may press `Ctrl` during `LinkHints.activate` (`f`) to toggle how it opens links, and `Shift`, `Alt` also change some points. You may see the current state of LinkHints at the bottom right corner of a browser window (named `HUD`).

**Comment by ghost** [comment]
Just to let you know, that `LinkHints.activateOpenInNewTab` acts as **Open a single link in Background**.
But `LinkHints.activateWithQueue` already satisfies this behaviour by allowing to **open multiple links in the background**. You can execute this by `<a-f>`, _open one link in bg_, then `<esc>`.
I assume this would be an optimized approach by removing `LinkHints.activateOpenInNewTab` isn't it? There are already too many **Open in New Tab** events, that populate F key combinations, which these two kinda feel duplicate.

For my purpose of instance, I consider `map F LinkHints.activateOpenInNewForegroundTab` to be the default.

**Reply by gdh1995** [reply]
Those names are kept to make Vimium C backwards compatible and them easier to notice and use for users.

Of course you may replace or remove those default mappings by adding customized `map` and `unmap` rules.
